utils/upload_ansible_roles_to_galaxy.py

```diff
@@ -82,132 +83,178 @@ def clone_and_init_repository(parent_dir, repo):
         os.chdir("..")
 
 
-def update_repository(repository, local_file_path):
-    print("Processing %s..." % repository.name)
-
-    with io.open(local_file_path, 'r', encoding="utf-8") as f:
-        filedata = f.read()
-
-    role_data = yaml.load(filedata)
-    vars_data = []
-    if "vars" in role_data[0]:
-        vars_data = role_data[0]["vars"]
-
-    tasks_data = []
-    if "tasks" in role_data[0]:
-        tasks_data = role_data[0]["tasks"]
-
-    # ansible language doesn't allow pre_tasks for roles, if the only pre task
-    # is the ansible version check we can ignore it because the minimal version
-    # is in role metadata
-    if "pre_tasks" in role_data[0]:
-        pre_tasks_data = role_data[0]["pre_tasks"]
-        if len(pre_tasks_data) == 1 and \
-                pre_tasks_data[0]["name"] == \
-                ssg.ansible.ansible_version_requirement_pre_task_name:
-            pass
-        else:
-            sys.stderr.write(
-                "%s contains pre_tasks other than the version check. "
-                "pre_tasks are not supported for ansible roles and "
-                "will be skipped!.\n")
-
-    tasks_local_content = yaml.dump(tasks_data, width=120, indent=4,
-                                    default_flow_style=False)
-
-    # Add \n in between tasks to increase readability
-    tasks_local_content = tasks_local_content.replace('\n- ', '\n\n- ')
-    tasks_remote_content = repository.get_file_contents("/tasks/main.yml")
-
-    if tasks_local_content != tasks_remote_content.decoded_content:
-        repository.update_file(
-            "/tasks/main.yml",
-            "Updates tasks/main.yml",
-            tasks_local_content,
-            tasks_remote_content.sha,
-            author=InputGitAuthor(
-                GIT_COMMIT_AUTHOR_NAME, GIT_COMMIT_AUTHOR_EMAIL)
-        )
-
-        print("Updating tasks/main.yml in %s" % repository.name)
-
-    vars_local_content = yaml.dump(vars_data, width=120, indent=4,
-                                   default_flow_style=False)
-    vars_remote_content = repository.get_file_contents("/vars/main.yml")
-
-    if vars_local_content != vars_remote_content.decoded_content:
-        repository.update_file(
-            "/vars/main.yml",
-            "Updates vars/main.yml",
-            vars_local_content,
-            vars_remote_content.sha,
-            author=InputGitAuthor(
-                GIT_COMMIT_AUTHOR_NAME, GIT_COMMIT_AUTHOR_EMAIL)
-        )
+class Repository(object):
+    def __init__(self, repo, local_filename):
+        self.repository = repo
```

Comment:
And this would be `remote_role` or `tracked_role`.

Reply:
I went for `remote_repo` - it is true that it is remote, but the type and usage of the variable are determined by the fact that it is a wrapper of a Github repository.